src/python/grpcio_observability/grpc_observability/observability_main.cc

```diff
@@ -0,0 +1,189 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/python/grpcio_observability/grpc_observability/observability_main.h"
+#include "src/python/grpcio_observability/grpc_observability/server_call_tracer.h"
+#include "src/python/grpcio_observability/grpc_observability/client_call_tracer.h"
+
+#include "absl/strings/string_view.h"
+
+namespace grpc_observability {
+
+std::queue<CensusData>* kCensusDataBuffer;
+std::mutex kCensusDataBufferMutex;
+std::condition_variable CensusDataBufferCV;
+// TODO(xuanwn): Change it to a more appropriate number
+constexpr int kExportThreshold = 2;
+
+
+void RecordIntMetric(MetricsName name, int64_t value, std::vector<Label> labels) {
+  Measurement measurement_data;
+  measurement_data.type = kMeasurementInt;
+  measurement_data.name = name;
+  measurement_data.value.value_int = value;
+
+  CensusData data = CensusData(measurement_data, labels);
+  AddCensusDataToBuffer(data);
+}
+
+
+void RecordDoubleMetric(MetricsName name, double value, std::vector<Label> labels) {
+  Measurement measurement_data;
+  measurement_data.type = kMeasurementDouble;
+  measurement_data.name = name;
+  measurement_data.value.value_double = value;
+
+  CensusData data = CensusData(measurement_data, labels);
+  AddCensusDataToBuffer(data);
+}
+
+
+void RecordSpan(SpanCensusData span_census_data) {
+  CensusData data = CensusData(span_census_data);
+  AddCensusDataToBuffer(data);
+}
+
+
+void NativeObservabilityInit() {
+    kCensusDataBuffer= new std::queue<CensusData>;
+}
+
+
+void* CreateClientCallTracer(char* method, char* trace_id, char* parent_span_id) {
+    void* client_call_tracer = new PythonOpenCensusCallTracer(method, trace_id, parent_span_id, PythonOpenCensusTracingEnabled());
+    return client_call_tracer;
+}
+
+
+void* CreateServerCallTracerFactory() {
+    void* server_call_tracer_factory = new PythonOpenCensusServerCallTracerFactory();
+    return server_call_tracer_factory;
+}
+
+
+void AwaitNextBatchLocked(std::unique_lock<std::mutex>& lock, int timeout_ms) {
+  auto now = std::chrono::system_clock::now();
+  CensusDataBufferCV.wait_until(lock, now + std::chrono::milliseconds(timeout_ms));
+}
+
+
+void AddCensusDataToBuffer(CensusData data) {
+  std::unique_lock<std::mutex> lk(kCensusDataBufferMutex);
+  kCensusDataBuffer->push(data);
+  if (kCensusDataBuffer->size() >= kExportThreshold) {
+      CensusDataBufferCV.notify_all();
+  }
+}
+
+
+GcpObservabilityConfig ReadObservabilityConfig() {
```

Comment:
This name implies that this is a [pure function](https://en.wikipedia.org/wiki/Pure_function), but it's modifying global mutable state. How about the name `ReadAndActivateObservabilityConfig`?